app/api/services/suppliers.py

```diff
@@ -43,6 +43,23 @@ def get_supplier_by_code(self, code):
             .one_or_none()
         )
 
+    def get_supplier_by_abn(self, abn):
```

Comment:
this should be get_supplier_by_abn.
since there are some instances where the abn is duplicated.
remove limit and change one_or_none to all()